### DISCUSSION TITLE
Any ideas how to set App Version for the client?

### DISCUSSION THREAD

**Discussion by igor-initech** [WhiskeySockets/Baileys#460] [General]
"Official" WhatsApp Web client connection shows up like this in Account Info (via Settings > Account > Request Account info):
`App Version: Web-2.2344.53`

However, Baileys client shows up like:
`App Version: MacOS-2.2344.53,MaybeUnofficial`
which is probably undesirable (draw attention / ban).
Any ideas how to control this in code?

**Comment by igor-initech** [comment]
Interestingly, the same details are sent back by WhatsApp during connection handshake, for example:
{"node":{"username":"79234299380","passive":true,"userAgent":{"platform":"MACOS","appVersion":{"primary":2,"secondary":2224,"tertiary":8},"mcc":"000","mnc":"000","osVersion":"0.1","manufacturer":"","device":"Desktop","osBuildNumber":"0.1","releaseChannel":"RELEASE","localeLanguageIso6391":"en","localeCountryIso31661Alpha2":"US"},"connectType":"WIFI_UNKNOWN","connectReason":"USER_ACTIVATED","device":36},"msg":"logging in..."}

I don't see any way to set userAgent in Baileys, any ideas?

**Comment by LuizLivio2** [comment]
> MacOS-2.2344.53,MaybeUnofficial

Where did you get this "MacOS-2.2344.53,MaybeUnofficial" return? I'm having a lot of numbers banned, and i think it might be something related to that

**Reply by igor-initech** [reply]
We solved it in our private fork:
![image](https://github.com/WhiskeySockets/Baileys/assets/2368764/6dbfaa83-1fac-44a6-8291-77b5b6ce6ccb)
But I don't think it really affects the ban rate.
